### PR TITLE
Update templates to use file scoped namespace declarations

### PR DESCRIPTION
**Pull request by DamianEdwards**
Depends on dotnet/roslyn#54763 being merged and flowing to SDK before dotnet/aspnetcore can get an SDK with support for this feature in the compiler.

Contributes to #33947 #33948

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -1,128 +1,120 @@
...
+            _logger = logger;
+        _downstreamWebApi = downstreamWebApi;
+    }
+
+    [HttpGet]
+    public async Task<IEnumerable<WeatherForecast>> Get()
+    {
+        HttpContext.VerifyUserHasAnyAcceptedScope(scopeRequiredByApi);
+
+        using var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:53]: @davidfowl does this trigger you?

**Review by pranavkm — APPROVED** [review]
As long as it compiles!

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs (hunk 1 above)
Comment (DamianEdwards) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:53]: For the record, I didn't change that, that's simply what was there already 🙃

Comment (davidfowl) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:53]: What the *&^%%$#@ is this

Comment (davidfowl) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:53]: Why is it using configure await, I have so many questions.

Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:53]: This was part of the Identity.Web work: https://github.com/dotnet/aspnetcore/issues/24024

**Comment by DamianEdwards** [comment]
So turns out the right branch of Roslyn wasn't flowing into the SDK so we would've been waiting a while for the feature to appear. That's now [been fixed](https://github.com/dotnet/sdk/commit/6dd2472dd83e6b8aae2947391ee3a59fe66934c2) so once code flow results in a new SDK being produced and flowed to installer which in turn flows to /aspnetcore (right?) we can finally get this change in.

**Comment by dougbu** [comment]
@DamianEdwards this is going to conflict w/ the move to https://github.com/dotnet/spa-templates and my current work on the second phase of #32027 (using the new repo as a submodule in this one). Any chance this could wait a day so that your changes to the Spa template could be made in the new repo❔

**Comment by DamianEdwards** [comment]
@DamianEdwards we're trying to get this in for preview.7, is your change targeting preview.7 too?

**Comment by dougbu** [comment]
> we're trying to get this in for preview.7, is your change targeting preview.7 too?

Yes❕ I'll put up the "phase 2" PR shortly after lunch I hope.

**Comment by DamianEdwards** [comment]
@dougbu so should I just undo the changes to the SPA templates given they're moving?

**Comment by dougbu** [comment]
> @dougbu so should I just undo the changes to the SPA templates given they're moving?

That'll work for me if it works for you.

**Comment by DamianEdwards** [comment]
Disabled the warnings that started showing up with the new SDK build (thanks to the new interpolated string builders language feature) on advice from @pranavkm.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**